controllers/k8sgpt_controller.go

```diff
@@ -133,6 +137,17 @@ func (r *K8sGPTReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return r.finishReconcile(nil, false)
 	}
 
+	if k8sgptConfig.Spec.AI.BackOff == nil {
```

Comment:
I am not sure if we want that behaviour when it's unspecified.
My mind says that if it's blank then we do nothing rather than enforcing again the default behaviour :thinking:  cause by having defaults in CRD means that an external factor/user changed the spec right?

I don't have a strong opinion though on this so eager to hear your thoughts @JuHyung-Son

Reply:
i expect if it is blank, it filled with be default `BackOff` values like other default string value.
i think in most cases, users may not explicitly define `BackOff` field.
If we don't update this field with default, users can not find default behavior by checking `yaml` in k9s. users should read documents, README carefully. but most users actually don`t read documents carefully.

So by filling default `BackOff` we can give hint how this CR works.
but we can do nothing and if `BackOff` is nil, do default behavior. but users may not explicitly know this.

i got this idea from k8s pod, when pod is created k8s inject serviaccount secret to pod. and explicitly shows it on CR, so users can check it through kubectl, k9s, lens ...

![CleanShot 2024-01-18 at 14 24 10](https://github.com/k8sgpt-ai/k8sgpt-operator/assets/20140126/76c92719-8bfc-4abc-b2be-951691e9612a)